synapseclient/core/credentials/credential_provider.py

```diff
@@ -144,6 +157,10 @@ def _get_auth_info(
                 "in the `~/.synapseConfig` file.",
             )
 
+        # If authToken is explicitly provided, return it directly and skip profiles
+        if user_login_args.auth_token:
+            return user_login_args.username, user_login_args.auth_token
+
```

Comment:
If we are going to provide this check, can we just do it right away so no other logic needs to be executed?